src/hover.ts

```diff
@@ -0,0 +1,99 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+'use strict';
+
+import * as vscode from 'vscode';
+import { debugChannel } from './utils/debug';
+import { RawInterval, HoverProperty } from './interfaces';
+
+export let hoverProvider: vscode.HoverProvider;
+export let hoverMap = new Map<string, Map<number, HoverProperty[]>>();
+
+export async function activate(_context: vscode.ExtensionContext) {
+  try {
+    hoverProvider = new JuvixHoverProvider();
+    vscode.languages.registerHoverProvider(
+      { language: 'Juvix', scheme: 'file' },
+      hoverProvider
+    )
+    debugChannel.info('Hover info registered');
+  } catch (error) {
+    debugChannel.error('No hover provider', error);
+  }
+}
+
+
+export class JuvixHoverProvider implements vscode.HoverProvider {
+  provideHover(document: vscode.TextDocument
+    , position: vscode.Position
+    , _token: vscode.CancellationToken
+  ) {
+    const filePath: string = document.fileName;
+    const line: number = position.line;
+    const col: number = position.character;
+    // debugChannel.info('Hover requested ------------------------');
+    // debugChannel.info(
+    //   'info',
+    //   'In file: ' + filePath + ' at: ' + (line + 1) + ':' + (col + 1)
+    // );
+    const hoversByFile = hoverMap.get(filePath);
+    if (!hoversByFile) {
+      // debugChannel.info(
+      //   'There is no hover info registered in file: ' + filePath
+      // );
+      return undefined;
+    }
+    const hoversByLine = hoversByFile.get(line);
+    if (!hoversByLine) {
+      // debugChannel.info(
+      //   'There is no definitions registered in line: ' + (line + 1)
+      // );
+      return undefined;
+    }
+
+    // debugChannel.info(
+    //   '> Found ' + hoversByLine.length + ' hovers at line: ' + (line + 1)
+    // );
+
+    for (let i = 0; i < hoversByLine.length; i++) {
+      const hoverProperty: HoverProperty = hoversByLine[i];
+      if (hoverProperty.interval.startCol <= col && col <= hoverProperty.interval.endCol) {
+        debugChannel.info('info', 'Hover text: ' + hoverProperty.text);
+        debugChannel.info('info', 'Hover interval: ' + JSON.stringify(hoverProperty.interval));
+        debugChannel.info('col', col.toString());
+        let enhancedText = new vscode.MarkdownString(
+          hoverProperty.text
+        );
+        enhancedText.isTrusted = true;
+        enhancedText.supportHtml = true;
+        let hover = new vscode.Hover(enhancedText, new vscode.Range(
+          new vscode.Position(hoverProperty.interval.line, hoverProperty.interval.startCol),
+          new vscode.Position(hoverProperty.interval.endLine, hoverProperty.interval.endCol)
+        )
+        );
+        return hover;
+      }
+    }
+    return undefined;
+  }
+}
+
+export function getHoverProperty(
+  entry: ((string | number)[] | string)[]
```

Comment:
Is this how `doc` or `goto` field may look like?
Should we create a separate type for that as it seems commonly used?

Reply:
No. For this hover property, we have:

  export interface HoverProperty {
    interval: RawInterval;
    text: string;
  }